src/svg.panzoom.js

```diff
@@ -27,20 +27,143 @@ extend(Svg, {
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMAX ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMAX ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMAX) {
+                viewportTopOffset = -heightOffset
+              }
+            }
+          } else if (preserveAspectRatio.meetOrSlice === preserveAspectRatio.SVG_MEETORSLICE_SLICE) {
+            // in slice mode, the viewport is the viewbox, shrunk on one
+            // or both sides to match the aspectRatio of the svg
+            if (changedAxis === 'width') {
+              const heightOffset = box.height - box.height * viewboxAspectRatio / svgAspectRatio
+              if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMAX) {
+                viewportTopOffset = heightOffset / 2
+                viewportBottomOffset = -heightOffset / 2
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMAX) {
+                viewportBottomOffset = -heightOffset
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMAX) {
+                viewportTopOffset = heightOffset
+              }
+            } else {
+              const widthOffset = box.width - box.width / viewboxAspectRatio * svgAspectRatio
+              if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMID) {
+                viewportLeftOffset = widthOffset / 2
+                viewportRightOffset = -widthOffset / 2
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMIN) {
+                viewportLeftOffset = -widthOffset
```

Comment:
Just looking at all the symmetry going on, this should be positive...

Reply:
Actually I think this should be `viewportRightOffset` and the other one `viewportLeftOffset`